### ISSUE TITLE
Named local functions inside method in MXML not emitted in JS

### ISSUE THREAD

**Issue by joshtynjala** [apache/royale-compiler#76]
## Code to reproduce

In MXML script block:

``` actionscript
private function methodWithLocalInMXML():void {
	function a(foo:int):void {};
	a(123);
}
```

## Actual output

``` js
MainView.prototype.MainView_methodWithLocalInMXML = function() {
  this.a(123);
};
```

## Expected output

```js
MainView.prototype.MainView_methodWithLocalInMXML = function() {
  var self = this;
  function a(foo) {
  };
  a(123);
};
```

## Other notes

* Works correctly in ActionScript classes, but not in MXML.
* This form is emitted correctly:
    ```actionscript
    var a = function(foo:int):void {};
    ```
* Commit f0ebbf0ca473a6938cb05f86b40d7053abab3014 is where local functions started being saved in FunctionNode and emitted at the top of the parent function.

**Comment by joshtynjala** [comment]
I haven't figured out exactly what's going wrong yet. The local function definitely gets saved by its parent function at some point. However, by the time that it gets emitted later, the parent reports that it has no local functions. This leads me to think that the parent might be recreated from scratch somewhere. Or it might be some kind of threading conflict where the collection storing local functions is replaced unexpectedly. Needs more investigation.

**Comment by aharui** [comment]
Script blocks in MXML are handled a bit differently than regular AS transpiling.  In regular AS, there is a ClassNode in the AST and the AST walk eventually gets down to functions with local functions.  In MXML ScriptBlocks there is no ClassNode in the AST and a ClassNode and/or ClassDefinition is assigned to the Emitter.  My guess is that the knowledge of local functions might be associated with something like a ClassNode but that association is invalid or broken when walking a Script block so the local function collection is lost to the code that needs it.

**Comment by joshtynjala** [comment]
Confirmed that a new FunctionNode is created for each function in an MXML script twice, and only once in AS3.

For a local function in AS3, the `EnumSet<PostProcessStep>` passed to `analyze()` contains `[CALCULATE_OFFSETS, POPULATE_SCOPE, RECONNECT_DEFINITIONS]`.

In MXML, the first time a local function is created, the call to `analyze()` contains `[CALCULATE_OFFSETS, POPULATE_SCOPE]` and the second time it's `[CALCULATE_OFFSETS, RECONNECT_DEFINITIONS]`.

Our call to rememberLocalFunction() happened only when the set contained POPULATE_SCOPE, so rememberLocalFunction() wasn't being called the second time, causing the local function to be... well, forgotten.

**Comment by joshtynjala** [comment]
Reopening because any calls to the local function are still emitted like `this.a(123);`
